### PR TITLE
70 navbar search button event handler & populate search results cards

### LINKED ISSUE #70: [IMPLEMENT] main.js - Write event handler for navbar search button to fetch API results for query and populate search results cards
# Feature Name

main.js - Write event handler for navbar search button to fetch API results for query and populate search results cards

# User Story

Our users would like to be able to search on our app.

Details: 
- add even handler to search button to fetch recipes for query 
- add recipes to localStorage 
- populate search result cards 
- navigate to search results page

### PR DESCRIPTION
**Pull request by hanran-y**
## Details 
add event handler to search button to fetch recipes for query
add recipes to localStorage
populate search result cards
fetch recipes when click show more button
## New Changes
- populate search result cards to search result page
- add show more button
- Fetch 10 recipes at once takes a lot of time(~1s), so I fetch 4 each search/show more
- add ID for button/class in index.html for convenient

## Linked Issue
Include issue number here
#70 
## Screenshots
Please include screenshots of any visual changes to the project (new HTML/CSS)

## Checklist
- [ ] I have requested 2+ code reviewers
- [ ] I have reviewed and commented my own code.
- [ ] My code follows the style guidelines of this project.
- [ ] I have added new tests as necessary.
- [ ] All new and existing tests pass successfully.

### CODE REVIEW

**Comment by hanran-y** [comment]
Not able to fetch data from API, Robert and me both tried today, I think is their server problem. I have contacted the spoonacular team, waiting for their response. More testing and "auto complete when no enough result" function will be implemented once I API work again.

File: source/index.html [hunk 1]
```diff
@@ -129,12 +129,16 @@ <h1 class="row-title" id="dinner">Top Dinner Recipes</h1>
         </div>
       </section>
 
-      <section class="search-results hidden">
+      <section class="search-results hidden" id="search-results-page">
         <!-- TODO: Populate query field -->
-        <h1>Top recipes for "bread"</h1>
-        <div class="results-grid">
+        <h1 id="search-results-title">Top recipes for "bread"</h1>
+        <div class="results-grid" id="results-grid-recipe-card-location">
```
Comment (farisashai) [inline source/index.html]: this id is really long can we make it simpler just like "search-results-container" or something

File: source/index.html [hunk 2]
```diff
@@ -129,12 +129,16 @@ <h1 class="row-title" id="dinner">Top Dinner Recipes</h1>
...
           <!-- TODO: Populate recipe cards -->
         </div>
+        <div class="button--wrapper">
+          <img src="source/images/arrow-down.png" alt="Arrow down" />
+          <button id="show-more-button">Show more</button>
```
Comment (farisashai) [inline source/index.html]: for now, just keep the button (no div or img). we can let antonia and jon style it after

File: source/js/main.js [hunk 3]
```diff
@@ -293,15 +293,90 @@ function initializeCreateRecipeButtons() {
...
+/**
+ * active when search button click
+ * store Recipe to local with query as category name
+ * Does NOT apply any filter & sort, only search by name
+ * 1.add even handler to search button to fetch recipes for query 
+ * 2.add fetched recipes to local Storage
+ * 3.populate recipe cards for search result page
+ * 4.TODO: if no enough result, call getRecipesByAutocomplete
+ */
+ async function searchButtonHandler() {
```
Comment (farisashai) [inline source/js/main.js]: I defined a function `openSearchResults` above already. please move all of the code for this function inside that one and leave the router.navigate as the last line there.

File: source/js/main.js [hunk 4]
```diff
@@ -293,15 +293,90 @@ function initializeCreateRecipeButtons() {
...
   // temporary code
   testCreateRecipeCards();
+  // localStorage.clear(); //for testing, turn off to make less api call
 }
 
 window.addEventListener('DOMContentLoaded', init);
+
+//search button event listener
+let sButton = document.getElementById('search-results-nav');
+sButton.addEventListener('click', () => searchButtonHandler());
```
Comment (farisashai) [inline source/js/main.js]: we don't need these two lines anymore if you move your code into the other function

File: source/js/main.js [hunk 5]
```diff
@@ -293,15 +293,90 @@ function initializeCreateRecipeButtons() {
...
   initializeRoutes();
   bindPopState();
   populateExplore();
   populateSavedRecipes();
   initializeCreateRecipeButtons();
-
+  
   // temporary code
   testCreateRecipeCards();
+  // localStorage.clear(); //for testing, turn off to make less api call
```
Comment (farisashai) [inline source/js/main.js]: we have a lot of unnecessary comments everywhere in our code. try to delete things like this before you commit them if we're not planning to keep them

**Review by farisashai — CHANGES_REQUESTED** [review]

File: source/index.html (hunk 1 above)
Comment (hanran-y) [reply source/index.html]: Renamed 'results-grid-recipe-card-location' to 'search-results-container', good naming

File: source/js/main.js (hunk 5 above)
Comment (hanran-y) [reply source/js/main.js]: roger that

File: source/js/main.js (hunk 3 above)
Comment (hanran-y) [reply source/js/main.js]: openSearchResults is not an async function, may I know how to use the async API fetcher inside this function?

Comment (farisashai) [reply source/js/main.js]: you can make it async just change the signature to 
```
const openSearchResults = async () => {
}
```

**Pull request merged**